core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -367,31 +360,9 @@ public static SqlNode convertConditionToSqlNode(RexNode node,
       joinContext =
           leftContext.implementor().joinContext(leftContext, rightContext);
       return joinContext.toSql(null, node);
-
-    case SEARCH:
-      joinContext =
-          leftContext.implementor().joinContext(leftContext, rightContext);
-      return joinContext.toSql(null, node);
```

Comment:
Is it possible to remove rest of the cases as well ?

Reply:
I tried to reduce the maximum cases but if I remove anyone now then getting many test failures on mig side.